spec/compiler/lexer/lexer_spec.cr

```diff
@@ -351,17 +367,17 @@ describe "Lexer" do
   it_lexes_u64 [["0xffff_ffff_ffff_ffff", "18446744073709551615"], ["0o177777_77777777_77777777", "18446744073709551615"], ["0b11111111_11111111_11111111_11111111_11111111_11111111_11111111_11111111", "18446744073709551615"]]
   it_lexes_u64 [["0x00ffffffffffffffff", "18446744073709551615"], ["0o001777777777777777777777", "18446744073709551615"], ["0b001111111111111111111111111111111111111111111111111111111111111111", "18446744073709551615"]]
   # 2**64
-  assert_syntax_error "0x10000_0000_0000_0000", "0x10000_0000_0000_0000 doesn't fit in an UInt64"
+  it_lexes_number :i128, ["0x10000_0000_0000_0000", "18446744073709551616"]
```

Comment:
From votes on https://github.com/crystal-lang/crystal/issues/8373#issuecomment-991891954 it is already quite clear that `Int128` literals should be possible to produce only with a suffix.

```suggestion
  it_lexes_number :i128, ["0x10000_0000_0000_0000_i128", "18446744073709551616"]
  assert_syntax_error "0x10000_0000_0000_0000", "0x10000_0000_0000_0000 doesn't fit in an UInt64; for Int128 add the i128 suffix"
```

Let's not change anything about `UInt64` (up to `2**64 - 1`) just yet, maybe the addition of a warning should be a separate PR and maybe it should come later.

Reply:
I used the error message "x doesn't fit in an UInt64, try using the suffix :i128" instead of the one given here, because a transition from alternative 4 to 3 seems likely and the error message would become way too long when two alternatives (u64 + i128) may be given